src/interface/state_descriptor.hpp

```diff
@@ -100,11 +100,19 @@ class StateDescriptor {
   StateDescriptor(const StateDescriptor &s) = delete;
 
   // Preferred constructor
-  explicit StateDescriptor(std::string const &label) : label_(label) {}
+  explicit StateDescriptor(std::string const &label) : label_(label) {
+    if (Metadata::FlagNameExists(label)) {
+      AddParam("MetadataFlag", Metadata::GetUserFlag(label));
```

Comment:
`MetadataFlag` is pretty ambiguous.  How about `MetadataPackageFlag` or similar?

Also, I don't see issue with it, but why are we going through `Param`s here? Is it only such that we can access the flag via "`MetadataFlag`" as follows
```c++
MetadataFlag GetMetadataFlag() { return params_.Get<MetadataFlag>("MetadataFlag"); }
```
hence bypassing the fact that the user flag actually takes on `label`?

Reply:
Updated the name.  Used `Param`s because it was convenient.  We use it for everything else after all.  Wanted the function that returns the flag in order to reduce the number of times anyone had to put things between double quotes that could introduce bugs via typos.